src/main/java/de/dennisguse/opentracks/ui/markers/MarkerEditActivity.java

```diff
@@ -176,7 +183,11 @@ protected void onCreate(Bundle savedInstanceState) {
 
     private Marker.Id createNewMarker(TrackRecordingService trackRecordingService) {
         try {
-            Marker.Id marker = trackRecordingService.insertMarker("", "", "", null);
+            if (location == null) {
+                throw new IllegalStateException("Location is null");
+            }
+            TrackPoint trackPoint = new TrackPoint(location, Instant.now());
```

Comment:
Trackpoints should only be created by trackpointcreator and during import. 
Rather pass a location (time is even not used for markers anymore).

Reply:
again, see first comment about Location vs. TrackPoint for Marker.